pymc3/variational/updates.py

```diff
@@ -0,0 +1,150 @@
+###########################################################
+#  Refactored by Dilin Wang
+#  Original implementation by Alec Radford, Luke Metz, Soumith Chintala: https://github.com/Newmu/dcgan_code   
+#  (c) Alec Radford, Luke Metz, Soumith Chintala
+###########################################################
+
+import theano
+import theano.tensor as tt
+import numpy as np
+
+
+__all__ = [
+    "SGD",
+    "Momentum",
+    "Adagrad",
+    "RMSprop",
+    "Adadelta",
+    "Adam"
+]
+
+def floatX(X):
+    return np.asarray(X, dtype=theano.config.floatX)
+
+class Update(object):
+
+    def __init__(self):
+        self.__dict__.update(locals())
```

Comment:
I'm still not satisfied about the current solution. 
1) Here you don't need init at all
2) locals have `self` and create circular reference